tests/unity/CMakeLists.txt

```diff
@@ -0,0 +1,176 @@
+  zephyr_link_libraries(${linker_str})
+endfunction()
+
+
+# Handle wrapping functions from mocked file.
+# Function takes header file and generates a file containing list of functions.
+# File is then passed to 'cmock_linker_trick' which adds linker option for each
+# function listed in the file.
+function(cmock_linker_wrap_trick header_file_path)
+  set(flist_file "${header_file_path}.flist")
+
+  execute_process(
+    COMMAND
+    ${PYTHON_EXECUTABLE}
+    ${NRF_DIR}/scripts/unity/func_name_list.py
+    --input ${header_file_path}
+    --output ${flist_file}
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    RESULT_VARIABLE op_result
+    OUTPUT_VARIABLE output_result
+  )
+
+  if (NOT ${op_result} EQUAL 0)
+    message(SEND_ERROR "${output_result}")
+    message(FATAL_ERROR "Failed to parse header ${header_file_path}")
+  endif()
+  cmock_linker_trick(${flist_file})
+endfunction()
+
+# Function takes original header and prepares two version
+# - version with system calls removed and static inline functions
+#   converted to standard function declarations
+# - version with addtional __wrap_ prefix for all functions that
+#   is used to generate cmock
+function(cmock_headers_prepare in_header out_header wrap_header)
+  execute_process(
+    COMMAND
+    ${PYTHON_EXECUTABLE}
+    ${NRF_DIR}/scripts/unity/header_prepare.py
+    "--input" ${in_header}
+    "--output" ${out_header}
+    "--wrap" ${wrap_header}
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    RESULT_VARIABLE op_result
+    OUTPUT_VARIABLE output_result
+  )
+
+  if (NOT ${op_result} EQUAL 0)
+    message(SEND_ERROR "${output_result}")
+    message(FATAL_ERROR "Failed to parse header ${in_header}")
+  endif()
+endfunction()
+
+#function for handling usage of mock
+#optional second argument can contain offset that include should be placed in
+#for example if file under test is include mocked header as <foo/header.h> then
+# mock and replaced header should be placed in <mock_path>/foo with <mock_path>
+# added as include path.
+function(cmock_handle header_file)
+  get_property(CMOCK_DIR GLOBAL PROPERTY CMOCK_DIR)
+  set(CMOCK_PRODUCTS_DIR ${APPLICATION_BINARY_DIR}/mocks)
+
+  #get optional offset macro
+  set (extra_macro_args ${ARGN})
+  list(LENGTH extra_macro_args num_extra_args)
+  if (${num_extra_args} EQUAL 1)
+    list(GET extra_macro_args 0 optional_offset)
+    set(dst_path "${CMOCK_PRODUCTS_DIR}/${optional_offset}")
+  else()
+    set(dst_path "${CMOCK_PRODUCTS_DIR}")
+  endif()
+
+  file(MAKE_DIRECTORY "${dst_path}/internal")
+
+  get_filename_component(header_name "${header_file}" NAME)
+  set(mod_header_path "${dst_path}/${header_name}")
+  set(wrap_header "${dst_path}/internal/${header_name}")
+
+  cmock_headers_prepare(${header_file} ${mod_header_path} ${wrap_header})
+  cmock_generate(${wrap_header} ${dst_path})
+
+  cmock_linker_wrap_trick(${mod_header_path})
+
+  target_include_directories(app PRIVATE ${CMOCK_PRODUCTS_DIR})
+  file(GLOB app_sources ${dst_path}/*.c)
```

Comment:
Globbing should be used with care.
The files under `${dst_path}` are all generated by the build system, correct ?
In that case I will assume that we ought to know somewhere else, exactly what files has been generated, and can use that information.

If that is not the case, then this code might be ok, as is.

Reply:
removed use of glob. Adding files explicitly.